### PR TITLE
checking for pip version

### PR DESCRIPTION
**Pull request by anukaal**
checking whether the pip has the latest version in the system or not

### CODE REVIEW

**Comment by story645** [comment]
What's the motivation for this PR?

**Comment by anukaal** [comment]
> Thank you for the PR (hi 👋), but what's the motivation for this PR? Basically, why should installing matplotlib via pip throw a warning to update pip?

to check whether pip version we are using is up to date in the system and we can install using pip !

File: setup.py [hunk 1]
```diff
@@ -53,7 +53,19 @@
     setupext.BackendMacOSX(),
     ]
 
-
+if sys.version_info < (3, 6):
```
Comment (tacaswell) [inline setup.py:56]: We will never get here because we will fail on L14 due to `if sys.version_info < py_min_version:` failing for any Python below 3.7.

**Comment by tacaswell** [comment]
Thank you for your work on this, however I am going to close without merging.  The check in https://github.com/matplotlib/matplotlib/blob/32630612e8d3c03a6092aeb7f2829c0fb35e61ee/setup.py#L11-L25 at the top of setup.py will catch every version of Python where the added logic would catch.

The reason that we have a comment about pip in that error message is that pip 9.0.1 is the first version of pip that understands the `python_requires` meta-data (https://github.com/matplotlib/matplotlib/blob/32630612e8d3c03a6092aeb7f2829c0fb35e61ee/setup.py#L315).  If you have an old version of Python and an old version of pip, when you try to install it will see a bunch of old versions of Matplotlib (with wheels for your old Python!) and a bunch of newer sdists.  Because old-pip does not know to look at `python_requires` it will enthusiastically grab the newest version of Matplotlib (which will not work with your Python) and try to compile it for you.  We give you that relatively nice error message (instead of a cryptic issue the first time you hit a too-new code construct or a compile error) as quickly as possible. 

I do not think we can do much more as at this point we do not _actually_ know we were invoked via pip, someone could have checked out the newer source and run `python setup.py`.

----

As a side note, the reason that we are still using `''.format` rather than fstrings is so that `setup.py` remains compatible with very old versions of Python for the purpose of providing that nice error message).

**Pull request closed without merging**